projects/stef.md

```diff
@@ -0,0 +1,330 @@
+# Otel/STEF
+
+## Background and description
+
+Otel/STEF (Sequential Tabular Encoding Format) is a new data format and network 
+protocol for OpenTelemetry data.
+
+On the tested datasets Otel/STEF payloads are several times more compact than OTLP, with 
+comparable serialization speed. Otel/STEF payloads are also smaller and serialization is 
+faster than Otel Arrow Phase 1.
+
+Otel/STEF targets a narrower niche than OTLP or Otel Arrow and is more efficient
+for that niche. Otel/STEF is optimized for payload size and fast serialization
+and is primarily intended to be used in the last communication leg from Otel Collector
+to observability backend where large volumes of data make network cost
+savings significant.
+
+Otel/STEF supports options that allow trading speed for size. At the highest 
+speed setting it can be faster than OTLP (_including_ conversion to/from pdata), while 
+producing significantly smaller payloads.
+
+A draft specification and a prototype implementation of Otel/STEF is attached to thi
+proposal.
+
+### Goals, objectives, and requirements
+
+The project goals are the following:
+
+- Implement missing capabilities in the [prototype implementation](#prototypes-and-draft-specs).
+- Extend benchmarking and test suite to cover more variety of datasets.
+- Turn the prototype into a production grade implementation.
+- Add support for traces and logs (the prototype implements metrics only).
+- Add Otel/STEF exporter and Otel/STEF receiver to Otel Collector.
+- Publish benchmark-justified guidelines on applicability of Otel/STEF vs OTLP vs Otel
+  Arrow.
+- Refine and stabilize Otel/STEF specification and encourage its implementation
+  inside or outside Otel.
+
+Project non-goals:
+
+- We do not plan to offer Otel/STEF as a general-purpose replacement for OTLP or for Otel
```

Comment:
I think I disagree with this. I understand the concern around limiting investment and engineering resources.   However - I think you need to better clarify the target use case in this situation.   Is this something *only* the Go SDK would be able to use with a collector?  Is this something for just Collector->Collector communication?   

That isn't really answered in this proposal, and I think it's critical.

My $.02 is that if we invest in a stateless protocol It should be an Open Standard (e.g. Arrow/parquet) *or* have an OpenTelemetry use case net well served by existing open standards and OTLP.

Here, you demonstrate that STEF can outperform arrow in efficiency for transmitting telemetry data.  **Where in open-telemetry do we use that?**

TL;DR; this should have a targeted set of use cases for the protocol.

Reply:
>have an OpenTelemetry use case net well served by existing open standards and OTLP.

This is my motivation. OTLP is inefficient size-wise. Otel Arrow Phase 1 is better size-wise, but even better is possible (as the benchmarks show) and is also quite expensive cpu-wise. STEF significantly advances the performance of our network protocols on multiple dimensions, with relatively small investment.

The first use case is as is described in the proposal: smaller wire size (network cost savings), less cpu consumption (compute cost savings) by the Collector. 

Do you think additional explanation is needed for this use case? Or do you want the additional use cases to be described (e.g. the mmap-ed direct-export)?

Reply:
I think we should showcase where stateful protocols are viable (e.g. how does it work through a load balancer?).

While I think stateful protocols have a lot of good use cases, from a broad sense - I have concerns that we may be optimising network overhead at the expense of inflexible network architecture and memory overhead in storing dictionaries from N clients.

I.e. for small, contextual cases this protocol is amazing and should be optimised for such cases.  In broad, highly distributed, cases, we may still need to invest in OTLP optimisations.

To be clear - I think STEF shows a lot of potential and is worth investing in.  I want to be explicit in this proposal where we think we see the biggest benefit and where the trade-offs fall off.  Let's have a target architecture in mind for benchmarks, comparison and 'success'.  I think you have that in your head (and were able to elucide when we talked about this), but it's just not written in the proposal.

Reply:
@jsuereth I completely agree with you, we need answers to that. I think this line in the list of goals touches that:

>Publish benchmark-justified guidelines on applicability of Otel/STEF vs OTLP vs Otel Arrow.

If this is not enough I can call it out more explicitly.

Reply:
Added more in https://github.com/open-telemetry/community/pull/2492/commits/bd6c8982dbc3e2e98ef4d6d0315d12fda89931ab